templates/user/dashboard/milestones.tmpl

```diff
@@ -34,34 +34,49 @@
-						{{svg "octicon-issue-closed" 16 "mr-3"}}
-						{{.i18n.Tr "repo.milestones.close_tab" .MilestoneStats.ClosedCount}}
-					</a>
-				</div>
-				<div class="ui right floated secondary filter menu">
-					<!-- Sort -->
-					<div class="ui dropdown type jump item">
-						<span class="text">
-							{{.i18n.Tr "repo.issues.filter_sort"}}
-							{{svg "octicon-triangle-down" 14 "dropdown icon"}}
-						</span>
-						<div class="menu">
-							<a class="{{if or (eq .SortType "closestduedate") (not .SortType)}}active{{end}} item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=closestduedate&state={{$.State}}">{{.i18n.Tr "repo.milestones.filter_sort.closest_due_date"}}</a>
-              <a class="{{if eq .SortType "furthestduedate"}}active{{end}} item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=furthestduedate&state={{$.State}}">{{.i18n.Tr "repo.milestones.filter_sort.furthest_due_date"}}</a>
-              <a class="{{if eq .SortType "leastcomplete"}}active{{end}} item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=leastcomplete&state={{$.State}}">{{.i18n.Tr "repo.milestones.filter_sort.least_complete"}}</a>
-              <a class="{{if eq .SortType "mostcomplete"}}active{{end}} item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=mostcomplete&state={{$.State}}">{{.i18n.Tr "repo.milestones.filter_sort.most_complete"}}</a>
-              <a class="{{if eq .SortType "mostissues"}}active{{end}} item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=mostissues&state={{$.State}}">{{.i18n.Tr "repo.milestones.filter_sort.most_issues"}}</a>
-              <a class="{{if eq .SortType "leastissues"}}active{{end}} item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=leastissues&state={{$.State}}">{{.i18n.Tr "repo.milestones.filter_sort.least_issues"}}</a>
+				<div class="ui three column stackable grid">
+					<div class="column">
+						<div class="ui compact tiny menu">
+							<a class="item{{if not .IsShowClosed}} active{{end}}" href="{{.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort={{$.SortType}}&state=open&q={{$.Keyword}}">
+								{{svg "octicon-issue-opened" 16 "mr-3"}}
+								{{.i18n.Tr "repo.milestones.open_tab" .MilestoneStats.OpenCount}}
+							</a>
+							<a class="item{{if .IsShowClosed}} active{{end}}" href="{{.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort={{$.SortType}}&state=closed&q={{$.Keyword}}">
+								{{svg "octicon-issue-closed" 16 "mr-3"}}
+								{{.i18n.Tr "repo.milestones.close_tab" .MilestoneStats.ClosedCount}}
+							</a>
 						</div>
 					</div>
-				</div>
-
+					<div class="column center aligned">
+						<form class="ui form ignore-dirty">
+							<div class="ui search fluid action input">
+								<input type="hidden" name="type" value="{{$.ViewType}}"/>
+								<input type="hidden" name="repos" value="[{{range $.RepoIDs}}{{.}}%2C{{end}}]"/>
```

Comment:
Somehow this value gets URL-encoded, which is not understood by gitea.
Maybe the backend should URI-decode properly, instead of fixing this on the frontend

Reply:
Hi, noerw,  I didn't make sense of your suggestion. What you mean is we should not use the encoding '%2c' in the URL query params on the frontend ?

I just reference it on the issue.tmpl, there is the same way to generate the query params. And the backend uses the regular express `var issueReposQueryPattern = regexp.MustCompile(`^\[\d+(,\d+)*,?\]$`)` to fetch them from URL query params.

Maybe we should use a unified way to solve it in the whole project?

Reply:
The issue is the double-escaped comma, the form handler that extracts the value from the hidden input already url-escapes.
To fix this:
```suggestion
								<input type="hidden" name="repos" value="[{{range $.RepoIDs}}{{.}},{{end}}]"/>
```

Reply:
replaced.